File: StarbuzzCoffee-my-version/src/abstract_class/Beverage.java
package abstract_class;

import interfaces.Condiment;

import java.util.List;

public abstract class Beverage {
    protected String description;
    protected double costCof;
    protected List condiments;

    public String getDescription(){
        return description;
    }

    public abstract void addCondiment(Condiment condiment);

    public abstract double cost();
}

File: src/main/java/developing/patterns/starbuzz-coffee/Beverage.kt
package developing.patterns.`starbuzz-coffee`

abstract class Beverage {
    open var description: String? = null
        protected set
    protected var costCof = 0.0
    protected var condiments: MutableList? = null

    abstract fun addCondiment(condiment: Condiment?)
    abstract fun cost(): Double
}